server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -558,6 +558,7 @@ public String resolveWriteIndexRouting(@Nullable String routing, String aliasOrI
             return routing;
         }
 
+        // TODO this is the same sort of code we have in bulk action already. we should share
```

Comment:
Over the years we've grown a lot of very similar code over in TransportBulkAction. We should unify these. *somehow*.